### PR TITLE
[pytorch] [build] Update for PyTorch v1.7.1 inference release

### PR DESCRIPTION
**Pull request by ydaiming**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ENV LD_LIBRARY_PATH /opt/conda/lib/:$LD_LIBRARY_PATH
+ENV PATH /opt/conda/bin:$PATH
+ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main
+ENV TEMP=/home/model-server/tmp
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends software-properties-common \
+    && add-apt-repository ppa:openjdk-r/ppa \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
```
Comment (saimidu) [inline pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu:25]: nit: Correct the indentation here.

File: pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+RUN pip uninstall -y torch \
+ && pip install --no-cache-dir -U $PT_INFERENCE_URL \
+ && pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U $PT_TORCHVISION_URL
+
+RUN conda install -y -c conda-forge pyyaml==5.3.1
+RUN pip install pillow==7.2.0 "awscli<2"
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
+
+RUN pip uninstall -y model-archiver mutli-model-server \
```
Comment (saimidu) [inline pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu:108]: Please fix the spelling of `multi-model-server`.

File: pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ && pip install --no-cache-dir -U $PT_INFERENCE_URL \
+ && pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U $PT_TORCHVISION_URL
+
+RUN conda install -y -c conda-forge pyyaml==5.3.1
+RUN pip install pillow==7.2.0 "awscli<2"
+
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
+
+RUN pip uninstall -y model-archiver mutli-model-server \
```
Comment (saimidu) [inline pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu:122]: Please fix the spelling of `multi-mode-server`

File: pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+ && pip install captum \
```
Comment (saimidu) [inline pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu:123]: `captum` is being installed twice - once in line 109, and here again. Please remove one of these.

File: pytorch/buildspec.yml [hunk 5]
```diff
@@ -35,9 +35,50 @@
         target: config.properties
 
   images:
-    BuildCPUPTTrainPy3DockerImage:
-      <<: *TRAINING_REPOSITORY
-      build: &PYTORCH_CPU_TRAINING_PY3 false
+    # BuildCPUPTTrainPy3DockerImage:
+    #   <<: *TRAINING_REPOSITORY
+    #   build: &PYTORCH_CPU_TRAINING_PY3 false
```
Comment (saimidu) [inline pytorch/buildspec.yml:40]: Please add the training images back to buildspec.

File: pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+ && pip install captum \
+ && pip install torchserve==$TS_VERSION \
+ && pip install torch-model-archiver==$TS_VERSION
+
+RUN cd tmp/ \
+ && rm -rf tmp*
```
Comment (saimidu) [inline pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu:114]: Why has this been added into the dockerfile?

File: pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+ && pip install torchserve==$TS_VERSION \
+ && pip install torch-model-archiver==$TS_VERSION
+
+RUN cd tmp/ \
+ && rm -rf tmp*
```
Comment (saimidu) [inline pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu:128]: Why has this been added into the dockerfile?

File: pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu (hunk 6 above)
Comment (ydaiming) [reply pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu:114]: There was a check failure regarding redundant files under the tmp folder.

File: pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu (hunk 7 above)
Comment (ydaiming) [reply pytorch/inference/docker/1.7.1/py3/cu110/Dockerfile.gpu:128]: Same as above.

File: pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu (hunk 6 above)
Comment (saimidu) [reply pytorch/inference/docker/1.7.1/py3/Dockerfile.cpu:113]: Where are the files in the `/tmp` folder coming from?

**Review by saimidu — APPROVED** [review]

**Pull request merged**